code/modules/tgs/tgs/v3210/api.dm

```diff
@@ -0,0 +1,253 @@
+#define REBOOT_MODE_NORMAL 0
+#define REBOOT_MODE_HARD 1
+#define REBOOT_MODE_SHUTDOWN 2
+
+#define SERVICE_WORLD_PARAM "server_service"
+#define SERVICE_INSTANCE_PARAM "server_instance"
+#define SERVICE_PR_TEST_JSON "prtestjob.json"
+#define SERVICE_INTERFACE_DLL "TGDreamDaemonBridge.dll"
+#define SERVICE_INTERFACE_FUNCTION "DDEntryPoint"
+
+#define SERVICE_CMD_HARD_REBOOT "hard_reboot"
+#define SERVICE_CMD_GRACEFUL_SHUTDOWN "graceful_shutdown"
+#define SERVICE_CMD_WORLD_ANNOUNCE "world_announce"
+#define SERVICE_CMD_LIST_CUSTOM "list_custom_commands"
+#define SERVICE_CMD_API_COMPATIBLE "api_compat"
+#define SERVICE_CMD_PLAYER_COUNT "client_count"
+
+#define SERVICE_CMD_PARAM_KEY "serviceCommsKey"
+#define SERVICE_CMD_PARAM_COMMAND "command"
+#define SERVICE_CMD_PARAM_SENDER "sender"
+#define SERVICE_CMD_PARAM_CUSTOM "custom"
+
+#define SERVICE_REQUEST_KILL_PROCESS "killme"
+#define SERVICE_REQUEST_IRC_BROADCAST "irc"
+#define SERVICE_REQUEST_IRC_ADMIN_CHANNEL_MESSAGE "send2irc"
+#define SERVICE_REQUEST_WORLD_REBOOT "worldreboot"
+#define SERVICE_REQUEST_API_VERSION "api_ver"
+
+#define SERVICE_RETURN_SUCCESS "SUCCESS"
+
+/datum/tgs_api/v3210
+	var/reboot_mode = REBOOT_MODE_NORMAL
+	var/comms_key
+	var/instance_name
+	var/originmastercommit
+	var/commit
+	var/list/cached_custom_tgs_chat_commands
+	var/warned_revison = FALSE
+	var/warned_custom_commands = FALSE
+
+/datum/tgs_api/v3210/ApiVersion()
+	return "3.2.1.0"
+
+/datum/tgs_api/v3210/proc/trim_left(text)
+	for (var/i = 1 to length(text))
+		if (text2ascii(text, i) > 32)
+			return copytext(text, i)
+	return ""
+
+/datum/tgs_api/v3210/proc/trim_right(text)
+	for (var/i = length(text), i > 0, i--)
+		if (text2ascii(text, i) > 32)
+			return copytext(text, 1, i + 1)
+	return ""
+
+/datum/tgs_api/v3210/proc/file2list(filename)
+	return splittext(trim_left(trim_right(file2text(filename))), "\n")
+
+/datum/tgs_api/v3210/OnWorldNew(datum/tgs_event_handler/event_handler, minimum_required_security_level)	//don't use event handling in this version
+	. = FALSE
+
+	comms_key = world.params[SERVICE_WORLD_PARAM]
+	instance_name = world.params[SERVICE_INSTANCE_PARAM]
+	if(!instance_name)
+		instance_name = "TG Station Server"	//maybe just upgraded
```

Comment:
Might want to change this

Reply:
This is for the instace name set inside TGS3 and is used for authing requests between the server and the tooling (See line 94). This is the instance named used after TGS3 has been upgraded and needs to re-attach to its instances. Its intentional.